woocommerce/wc-admin
# Remove tasks feature flag and old components

Fixes #7374 

* Enables the task list components in all environments and cleans up old components.
* Moves completion checks and tracks to `Task` and `TaskList` models.
* Updates the task `isComplete` to check for all finished tasks completed instead of previously tracked completion (this could use a quick sanity check to make sure this is the information we want).

### Detailed test instructions:

1. Switch to this branch (make sure to `npm start` again to refresh the feature flags).
2. Smoke test the task list.
3. Delete the options `woocommerce_task_list_completed_lists` and `woocommerce_task_list_tracked_completed_tasks`
4. Load the task list.
5. Make sure the above options now contain the correct completed items (list IDs and task IDs respectively)
